codegen-server/typescript/src/main/kotlin/software/amazon/smithy/rust/codegen/server/typescript/smithy/generators/TsApplicationGenerator.kt

```diff
@@ -0,0 +1,303 @@
+                    ))
+                    """.trimIndent(),
+                    *codegenScope,
+                )
+            }
+
+            writer.rust(
+                """
+                /// Clone the inner socket allowing it to be shared between multiple
+                /// Nodejs processes.
+                """.trimIndent(),
+            )
+            Attribute("napi").render(writer)
+            writer.rustBlockTemplate(
+                // XXX Can I change the signature to #{napi}::Result<Self>?
+                """pub fn try_clone(&self) -> #{napi}::Result<TsSocket>""".trimIndent(),
+                *codegenScope,
+            ) {
+                writer.rustTemplate(
+                    """
+                    Ok(TsSocket(
+                        self.0
+                            .try_clone()
+                            .map_err(|e| #{napi}::Error::from_reason(e.to_string()))?,
+                    ))
+                    """.trimIndent(),
+                    *codegenScope,
+                )
+            }
+        }
+
+        writer.rustBlock("impl TsSocket") {
+            writer.rustBlockTemplate(
+                """pub fn to_raw_socket(&self) -> #{napi}::Result<#{socket2}::Socket>""".trimIndent(),
+                *codegenScope,
+            ) {
+                writer.rustTemplate(
+                    """
+                    self.0
+                     .try_clone()
+                     .map_err(|e| #{napi}::Error::from_reason(e.to_string()))
+                    """.trimIndent(),
+                    *codegenScope,
+                )
+            }
+        }
+    }
+
+    private fun renderServer(writer: RustWriter) {
+        writer.rustBlockTemplate(
+            // / XXX Check here the aws_smithy_http_server (same issue with new_socket above)
+            """pub fn start_hyper_worker(
+            socket: &TsSocket,
+            app: #{tower}::util::BoxCloneService<
+                #{http}::Request<aws_smithy_http_server::body::Body>,
+                #{http}::Response<aws_smithy_http_server::body::BoxBody>,
+                std::convert::Infallible,
+            >,
+                ) -> #{napi}::Result<()>
+            """.trimIndent(),
+            *codegenScope,
+        ) {
+            // XXX Check here IntoMakeService
+            writer.rustTemplate(
+                """
+                let server = #{hyper}::Server::from_tcp(
+                        socket
+                            .to_raw_socket()?
+                            .try_into()
+                            .expect("Unable to convert socket2::Socket into std::net::TcpListener"),
+                )
+                .expect("Unable to create hyper server from shared socket")
+                .serve(aws_smithy_http_server::routing::IntoMakeService::new(app));
+
+                // TODO(https://github.com/awslabs/smithy-rs/issues/2317) Find a better, non-blocking way
+                // to spawn the server.
```

Comment:
This is already the better, non blocking way to spawn the server. I don't understand this comment.